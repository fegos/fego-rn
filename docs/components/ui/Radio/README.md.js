webpackJsonp([13],{489:function(t,a){t.exports={content:["article",["h1","\u5355\u9009\u6846"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u5355\u9009\u9009\u62e9\u6846"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"js",highlighted:'<span class="token operator">&lt;</span>Radio disabled<span class="token operator">=</span><span class="token punctuation">{</span><span class="token boolean">true</span><span class="token punctuation">}</span><span class="token operator">></span>disabled<span class="token operator">&lt;</span><span class="token operator">/</span>Radio<span class="token operator">></span>\n\n<span class="token operator">&lt;</span>Radio defaultChecked<span class="token operator">=</span><span class="token punctuation">{</span><span class="token boolean">false</span><span class="token punctuation">}</span><span class="token operator">></span>\u975e\u53d7\u63a7\u7ec4\u4ef6\uff1a\u4f7f\u7528defaultChecked<span class="token operator">&lt;</span><span class="token operator">/</span>Radio<span class="token operator">></span>\n\n<span class="token operator">&lt;</span>Radio checked<span class="token operator">=</span><span class="token punctuation">{</span><span class="token keyword">this</span><span class="token punctuation">.</span>state<span class="token punctuation">.</span>checked<span class="token punctuation">}</span> onChange<span class="token operator">=</span><span class="token punctuation">{</span><span class="token punctuation">(</span>checked<span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span>\n\t<span class="token keyword">this</span><span class="token punctuation">.</span><span class="token function">setState</span><span class="token punctuation">(</span><span class="token punctuation">{</span>\n\t\tchecked<span class="token punctuation">:</span> checked\n\t<span class="token punctuation">}</span><span class="token punctuation">)</span>\n<span class="token punctuation">}</span><span class="token punctuation">}</span><span class="token operator">></span>\u53d7\u63a7\u7ec4\u4ef6\uff1a\u4f7f\u7528checked<span class="token operator">&lt;</span><span class="token operator">/</span>Radio<span class="token operator">></span>'},["code","<Radio disabled={true}>disabled</Radio>\n\n<Radio defaultChecked={false}>\u975e\u53d7\u63a7\u7ec4\u4ef6\uff1a\u4f7f\u7528defaultChecked</Radio>\n\n<Radio checked={this.state.checked} onChange={(checked)=>{\n\tthis.setState({\n\t\tchecked: checked\n\t})\n}}>\u53d7\u63a7\u7ec4\u4ef6\uff1a\u4f7f\u7528checked</Radio>"]],["h2","API"],["h3","props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","checked"],["td","\u53d7\u63a7\u5c5e\u6027\uff1a\u9700\u914d\u5408 onChange \u4f7f\u7528\u66f4\u65b0\u6570\u636e checked"],["td","boolean"],["td","-"]],["tr",["td","defaultChecked"],["td","\u975e\u63a7\u5c5e\u6027\uff1a\u9ed8\u8ba4\u9009\u4e2d\u72b6\u6001"],["td","boolean"],["td","false"]],["tr",["td","disabled"],["td","\u662f\u5426\u7981\u6b62\u9009\u62e9\u6846"],["td","boolean"],["td","false"]],["tr",["td","iconFamily"],["td","\u56fe\u6807\u5b57\u4f53\u540d"],["td","string"],["td","-"]],["tr",["td","iconCheckName"],["td","\u9009\u4e2d\u56fe\u6807\u540d\u79f0"],["td","string"],["td","check-square-o"]],["tr",["td","iconUncheckName"],["td","\u672a\u9009\u4e2d\u56fe\u6807\u540d\u79f0"],["td","string"],["td","square-o"]],["tr",["td","onChange"],["td","\u9009\u62e9\u6846\u53d8\u5316\u56de\u8c03\uff0c\u5f53\u4f7f\u7528 checked \u5c5e\u6027\u65f6\u9700\u914d\u5408 onChange \u66f4\u65b0\u6570\u636echecked \u5c5e\u6027"],["td","function"],["td","-"]],["tr",["td","left"],["td","\u56fe\u6807\u4f4d\u7f6e"],["td","boolean"],["td","true"]]]],["h3","\u6837\u5f0f\u5bf9\u8c61styles"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u9002\u7528\u7c7b\u578b"]]],["tbody",["tr",["td","container"],["td","\u5bb9\u5668\u6837\u5f0f"],["td","View"]],["tr",["td","icon"],["td","\u56fe\u6807\u6837\u5f0f"],["td","Icon"]],["tr",["td","text"],["td","\u9009\u62e9\u6846\u6587\u5b57\u6837\u5f0f"],["td","Text"]],["tr",["td","checked"],["td","\u9009\u4e2d\u65f6\u7684\u6837\u5f0f"],["td","Icon"]],["tr",["td","disabled"],["td","\u7981\u6b62\u65f6\u7684\u6837\u5f0f"],["td","Icon\uff0cText"]]]]],meta:{title:"Radio",subTitle:"\u5355\u9009\u6846\uff0c\u4e00\u822c\u4e0eGroup\u914d\u5408\u4f7f\u7528\u5b9e\u73b0\u5355\u9009",filename:"components/ui/Radio/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u5355\u9009\u6846",title:"\u5355\u9009\u6846"},"\u5355\u9009\u6846"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#props\u5217\u8868",title:"props\u5217\u8868"},"props\u5217\u8868"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u6837\u5f0f\u5bf9\u8c61styles",title:"\u6837\u5f0f\u5bf9\u8c61styles"},"\u6837\u5f0f\u5bf9\u8c61styles"]]]}}});