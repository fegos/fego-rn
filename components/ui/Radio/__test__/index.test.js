import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Radio from '../index';

describe('Radio item', () => {
  it('renders correctly', () => {
    const tree = Renderer.create((<Radio />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('press event', () => {
    const wrapper = shallow(<Radio>我是一个Radio</Radio>);
    expect(wrapper.instance().state.checked).toBe(false);
    wrapper.simulate('press');
    expect(wrapper.instance().state.checked).toBe(true);
  });

  it('checked props', () => {
    const handler = jest.fn();
    const wrapper = shallow(<Radio checked={false} onChange={handler}>使用受控属性checked的Radio</Radio>);
    wrapper.simulate('press');
    expect(handler).toBeCalled();
  });

  it('wrapper name', () => {
    const wrapper = shallow(<Radio>Radio</Radio>);
    expect(wrapper.name()).toMatch(/TouchableWithoutFeedback/);
  });
});
