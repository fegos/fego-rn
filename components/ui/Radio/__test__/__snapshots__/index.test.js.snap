// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio item renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Array [
          Object {
            "color": "#333",
            "fontSize": 16,
            "paddingRight": 0,
            "width": 20,
          },
          null,
          null,
        ],
        Object {},
        Object {
          "fontFamily": "FontAwesome",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
